Sources/AuthFoundation/Token Management/IDTokenValidator.swift

```diff
@@ -36,4 +36,7 @@ public protocol IDTokenValidatorContext {
     
     /// The maximum age the token should support when authenticating.
     var maxAge: TimeInterval? { get }
+    
+    /// The possible option using  in validating a claim
+    var validationOptions: ValidationOption? { get }
```

Comment:
All of these options are required by the spec. https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation

If someone wants to break the spec, I don't mind forcing pain on them (by having them reimplement this class, likely via copy/paste and modifying the lines they want changed).

Reply:
I agree, this is definitely something we need to think about. I think in this case the spec defines the order/manner for the validation. _(Which brings a valid question of why make it modifiable by a client.)_ . My opinion is we can guarantee the order at which the validation occurs and provide the flexibility on the type of validation that occurs _(if someone has a use case that requires skipping a check)_

Reply:
All good points. I was thinking about future-proofing for the real-world issues that some developers face, but at the outset we shouldn't optimize for bypassing security checks. If anything, the original intent of this PR (to simplify the readability and testability of validation checks) still stands. If we need to make this customizable in the future, this gives us a path forward to do that without a lot of effort.